api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -192,6 +194,10 @@ public void registerUser(String contactEmail, String firstName, String lastName)
 
   @Override
   public void createAllOfUsBillingProject(String projectName) {
+    if (projectName.contains(WORKSPACE_DELIMITER)) {
+      throw new RuntimeException("Attempting to create billing project with name that contains workspace delimiter : " + projectName);
```

Comment:
Maybe just log severe instead. This wouldn't be the end of the world, and I'm not sure we want to be 500ing requests at runtime for this reason.

Reply:
hmm, I think throwing an exception is correct here since it should really never happen. There is no code path that should allow it at the moment and if a developer makes a change so that it does happen, it is likely because they were unaware of how it interacts with the delimiter. In that case, I think it should fail quickly and apparently so we can detect the change before it gets merged.

Reply:
I'm not convinced this would be caught before merging since the prefix is config-driven and grabbed at runtime, but I guess this at least covers the suffix (alternatively, you could check for this in a test instead).

IllegalArgumentException is probably better if you leave the assertion here.